src/utils/index.js

```diff
@@ -97,6 +100,45 @@ module.exports = {
     }
   },
 
+  getChainId: async (ethereumProvider) => {
+    const method = 'eth_chainId'
+    const params = []
+
+    const chainIdHex = await safeEthSend(ethereumProvider, {
+      jsonrpc: '2.0',
+      id: 0,
+      method,
+      params
+    })
+    return parseInt(chainIdHex, 16)
+  },
+
+  getCode: async (ethereumProvider, address) => {
+    const method = 'eth_getCode'
+    const params = [address]
+
+    const code = await safeEthSend(ethereumProvider, {
+      jsonrpc: '2.0',
+      id: 1,
+      method,
+      params
+    })
+    return code
+  },
+
+  isValidSignature: async (linkObj, isErc1271, web3Provider) => {
+    if (!linkObj.address) return false
+    if (!isErc1271) return true
```

Comment:
Why always return true here?

Reply:
The only check we're doing for EOA is the address extraction before we call the `isValidSignature`.  Maybe we could rewrite this into:
```javascript
if (!isErc1271) {
  if (!linkObj.address) return false
  return true
}
```

Reply:
We could event extract this into `isValidEOALink` and `isValid1271Link` and check the input params in both methods